services/provider/server/server.go

```diff
@@ -1042,3 +1043,143 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
 
 	return &pb.PeerStorageClusterResponse{}, nil
 }
+
+func (s *OCSProviderServer) GetStorageClientsInfo(ctx context.Context, req *pb.StorageClientsInfoRequest) (*pb.StorageClientsInfoResponse, error) {
+	var clientsInfo []*pb.ClientInfo
+	var clientError []*pb.StorageClientInfoError
+
+	for i := range req.ClientIDs {
+		consumer, err := s.consumerManager.Get(ctx, req.ClientIDs[i])
+		if err != nil {
+			klog.Errorf("failed to get consumer with id %v: %v", req.ClientIDs[i], err)
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  err.Error(),
+				},
+			)
+			continue
+		}
+		owner := util.FindOwnerRefByKind(consumer, "StorageCluster")
+
+		if owner == nil {
+			klog.Errorf("failed to find StorgeCluster owning the StorageConsumer %v", req.ClientIDs[i])
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "failed to find StorgeCluster owning the StorageConsumer",
+				},
+			)
+			continue
+		}
+
+		if owner.UID != types.UID(req.StorageClusterUID) {
+			klog.Errorf("storageCluster specified on the req does not own the Client")
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_InvalidArgument,
+					Message:  "storageCluster specified on the req does not own the Client",
+				},
+			)
+			continue
+		}
+
+		rnsList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err = s.client.List(
+			ctx,
+			rnsList,
+			client.InNamespace(s.namespace),
+			client.MatchingLabels{controllers.StorageConsumerNameLabel: consumer.Name},
+			client.Limit(1),
+		)
+		if err != nil {
+			klog.Error(err)
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "rados namespace for the client not found",
+				},
+			)
+			continue
+		}
```

Comment:
Not found is not considered an error. it just means we cannot return a rados namespace name

Reply:
Here, there is an error fetching the radosnamespace, maybe I need to change the error Message

Reply:
If a client has not issued a rados namespace yet (because the other controller is broken or just delayed) it does not mean the client does not exists or that we should not return info for it